h2/src/test/org/h2/test/unit/TestIgnoreCatalogs.java

```diff
@@ -0,0 +1,158 @@
+package org.h2.test.unit;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.h2.test.TestBase;
+import org.h2.test.TestDb;
+
+/**
+ * @author aschoerk
+ */
+public class TestIgnoreCatalogs extends TestDb {
+    /**
+     * Run just this test.
+     *
+     * @param a ignored
+     */
+    public static void main(String... a) throws Exception {
+        TestBase.createCaller().init().test();
+    }
+
+    @Override
+    public void test() throws Exception {
+        canUseDefaultSchema();
+        canYetIdentifyWrongCatalogName();
+        canUseSettingInUrl();
+        canUseSetterSyntax();
+        canCatalogNameEqualSchemaName();
+        canUseCatalogAtIndexName();
+        canCommentOn();
+        canAllCombined();
+    }
+
+    public void canCommentOn() throws Exception {
+        try (Connection conn = getConnection("jdbc:h2:mem:testdb;IGNORE_CATALOGS=TRUE;init=drop all objects\\;create schema dbo\\;set schema dbo\\;")) {
+            try (Statement stat = conn.createStatement()) {
+                stat.execute("create table catalog1.dbo.test(id int primary key, name varchar(255))");
+                stat.execute("comment on table catalog1.dbo.test is 'table comment1'");
+                stat.execute("comment on table dbo.test is 'table comment2'");
+                stat.execute("comment on table catalog1..test is 'table comment3'");
+                stat.execute("comment on table test is 'table comment4'");
+                stat.execute("comment on column catalog1.dbo.test.id is 'id comment1'");
+                stat.execute("comment on column dbo.test.id is 'id comment1'");
+                stat.execute("comment on column catalog1..test.id is 'id comment1'");
+                stat.execute("comment on column test.id is 'id comment1'");
+            }
+        }
+    }
+
+    public void canUseDefaultSchema() throws Exception {
+        try (Connection conn = getConnection("jdbc:h2:mem:testdb;IGNORE_CATALOGS=TRUE;init=drop all objects\\;create schema dbo\\;set schema dbo\\;")) {
+            try (Statement stat = conn.createStatement()) {
+                stat.execute("create table catalog1..test(id int primary key, name varchar(255))");
+
+                stat.execute("create table test2(id int primary key, name varchar(255))");
+                // expect table already exists
+                assertThrows(42101, stat, "create table catalog2.dbo.test(id int primary key, name varchar(255))");
```

Comment:
Do not use codes here and in other places, use the constants from the `ErrorCode` instead.

Reply:
fixed